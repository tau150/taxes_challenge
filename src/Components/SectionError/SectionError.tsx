import { Alert, AlertIcon, AlertTitle, AlertDescription } from "@chakra-ui/react";

interface Props {
  title: string;
  description: string;
}
const SectionError = ({ title, description }: Props) => {
  return (
    <Alert
      alignItems="center"
      flexDirection="column"
      height="200px"
      justifyContent="center"
      status="error"
      textAlign="center"
      variant="subtle"
    >
      <AlertIcon boxSize="40px" mr={0} />
      <AlertTitle fontSize="lg" mb={1} mt={4}>
        {title}
      </AlertTitle>
      <AlertDescription maxWidth="sm">{description}</AlertDescription>
    </Alert>
  );
};

export default SectionError;
