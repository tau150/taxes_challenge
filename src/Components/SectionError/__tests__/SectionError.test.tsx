import { expect } from "vitest";

import SectionError from "../SectionError";

import { render, screen } from "tests/render";

describe("SectionError component", () => {
  it("should render properly with default values", () => {
    const { container } = render(<SectionError />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render properly with title and description provided ", () => {
    const FAKE_TITLE = "some title";
    const FAKE_DESCRIPTION = "some description";

    render(<SectionError description={FAKE_DESCRIPTION} title={FAKE_TITLE} />);

    expect(screen.getByText(FAKE_TITLE)).toBeInTheDocument();
    expect(screen.getByText(FAKE_DESCRIPTION)).toBeInTheDocument();
  });
});
