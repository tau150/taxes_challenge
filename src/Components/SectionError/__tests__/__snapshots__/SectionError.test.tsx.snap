// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SectionError component > should render properly with default values 1`] = ` `;
